packages/artillery/lib/platform/aws-ecs/legacy/run-cluster.js

```diff
@@ -978,19 +992,83 @@ async function createTestBundle(context) {
       {
         name: context.testId,
         config: context.cliOptions.config,
-        packageJsonPath: context.packageJsonPath
+        packageJsonPath: context.packageJsonPath,
+        flags: context.cliOptions
       },
-      function (err, _result) {
+      function (err, result) {
         if (err) {
           return reject(err);
         } else {
+          context.fullyResolvedConfig = result.manifest.fullyResolvedConfig;
           return resolve(context);
         }
       }
     );
   });
 }
 
+async function createADOTDefinitionIfNeeded(context) {
```

Comment:
As we agreed @hassy , I'm filtering the `publish-metrics` config in a separate function here before resolving the config and env vars for ADOT.

However maybe it can be moved into the `resolveADOTConfigSettings` to allow for better separation of concerns:  everything `publish-metrics` related would get done on `publish-metrics` side , and this function only does `run-cluster` type stuff (ssm, task definition, etc) ?  Let me know.